docs/contributor/writing-tests-for-vscode-swift.md

```diff
@@ -113,6 +114,30 @@ suite("ReloadExtension Unit Test Suite", () => {
 
 You may have also noticed that we needed to cast the `"Reload Extensions"` string to `any` when resolving `showWarningMessage()`. Unforunately, this may be necessary for methods that have incompatible overloaded signatures due to a TypeScript issue that remains unfixed.
 
+## Mocking the File System
+
+The [`mock-fs`](https://github.com/tschaub/mock-fs) module can be used to temporarily replace Node's built-in `fs` module with an in-memory file system. This can be useful for testing logic that uses the `fs` module without actually reaching out to the file system. Just make sure that you add a `teardown()` block that restores the `fs` module after each test:
+
+```typescript
+import * as chai from "chai";
+import * as mockFS from "mock-fs";
+import * as fs from "fs/promises";
```

Comment:
let's go over this with group. Not clear where the utilities we have are lacking that we need to add a new dependency. Not saying we shouldn't do this, but should make clear the rational since this needs to be adopted for the project's strategy

Reply:
Fair. The problem with our current setup is that there are multiple ways to test for files existing. You could do `fs.readFile()` and catch failures, or you could do the same with `fs.stat()`. The `mock-fs` module lets you declare in your test which files exist under which directories and backs the entirety of the `fs` module with an in-memory filesystem. This means that the test is more resilient to changes in implementation.

Reply:
I've added a paragraph here to talk about why `mock-fs` is necessary.